classes/core/PKPApplication.inc.php

```diff
@@ -194,83 +194,15 @@ public function initializeDatabaseConnection() {
 		if ($databaseConnectionInitialized) return;
 
 		$databaseConnectionInitialized = true;
-		$laravelContainer = new Illuminate\Container\Container();
-		Registry::set('laravelContainer', $laravelContainer);
-		(new Illuminate\Bus\BusServiceProvider($laravelContainer))->register();
-		(new Illuminate\Events\EventServiceProvider($laravelContainer))->register();
-		$eventDispatcher = new \Illuminate\Events\Dispatcher($laravelContainer);
-		$laravelContainer->instance('Illuminate\Contracts\Events\Dispatcher', $eventDispatcher);
-		$laravelContainer->instance('Illuminate\Contracts\Container\Container', $laravelContainer);
-
-		// Map valid config options to Illuminate database drivers
-		$driver = strtolower(Config::getVar('database', 'driver'));
-		if (substr($driver, 0, 8) === 'postgres') {
-			$driver = 'pgsql';
-		} else {
-			$driver = 'mysql';
-		}
-
-		$capsule = new Capsule;
-		$capsule->addConnection([
-			'driver'    => $driver,
-			'host'      => Config::getVar('database', 'host'),
-			'database'  => Config::getVar('database', 'name'),
-			'username'  => Config::getVar('database', 'username'),
-			'port'      => Config::getVar('database', 'port'),
-			'unix_socket'=> Config::getVar('database', 'unix_socket'),
-			'password'  => Config::getVar('database', 'password'),
-			'charset'   => Config::getVar('i18n', 'connection_charset', 'utf8'),
-			'collation' => Config::getVar('database', 'collation', 'utf8_general_ci'),
-		]);
-		$capsule->setEventDispatcher($eventDispatcher);
-		$capsule->setAsGlobal();
-		if (Config::getVar('database', 'debug')) Capsule::listen(function($query) {
-			error_log("Database query\n$query->sql\n" . json_encode($query->bindings));//\n Bindings: " . print_r($query->bindings, true));
-		});
-
-
-		// Set up Laravel queue handling
-		$laravelContainer->bind('exception.handler', function () {
```

Comment:
I think this was required for the queuing code to work -- see e.g. https://github.com/pkp/pkp-lib/blob/main/api/v1/_email/PKPEmailHandler.inc.php#L207 -- and was added in https://github.com/pkp/pkp-lib/issues/4017. Are you sure this isn't needed any more?

Reply:
(Perhaps this is now provided by `queue.failer`?)

Reply:
Ideally, `PKPEmailHandler` requires refactoring, where that part with the worker initialization is removed in favor of creating a Job class. Laravel already registered a worker at this point. See: `Illuminate\Queue\QueueServiceProvider::registerWorker()`. As an intermediate solution, the worker expects the `Illuminate\Contracts\Debug\ExceptionHandler` as a 3rd parameter for the constructor, so can just feed it to it. 

In order to initialize `queue.failer`, I need to pass the data required to its initialization in the config (probably only table name but taking a look).

Reply:
Indeed, looks like Laravel doesn't do the binding between ExceptionHandler contract and its implementation on the Container or Service Provider level. Re-added this.